ddl/ddl_api.go

```diff
@@ -3186,3 +3230,105 @@ func buildPartitionInfo(meta *model.TableInfo, d *ddl, spec *ast.AlterTableSpec)
 	}
 	return part, nil
 }
+
+// LockTables uses to execute lock tables statement.
+func (d *ddl) LockTables(ctx sessionctx.Context, stmt *ast.LockTablesStmt) error {
+	lockTables := make([]model.TableLockTpInfo, 0, len(stmt.TableLocks))
+	is := d.infoHandle.Get()
+	sessionInfo := model.SessionInfo{
+		ServerID:  d.GetID(),
+		SessionID: ctx.GetSessionVars().ConnectionID,
+	}
+	uniqueTableID := make(map[int64]struct{})
+	// Check whether the table was already locked by other.
+	for _, tl := range stmt.TableLocks {
+		tb := tl.Table
+		// TODO: replace const string "performance_schema" with xxx.LowerName.
+		// Currently use perfschema.LowerName will have import cycle problem.
+		if tb.Schema.L == infoschema.LowerName || tb.Schema.L == "performance_schema" || tb.Schema.L == mysql.SystemDB {
+			if ctx.GetSessionVars().User != nil {
+				return infoschema.ErrAccessDenied.GenWithStackByArgs(ctx.GetSessionVars().User.Username, ctx.GetSessionVars().User.Hostname)
+			}
+			return infoschema.ErrAccessDenied
+		}
+		schema, ok := is.SchemaByName(tb.Schema)
+		if !ok {
+			return infoschema.ErrDatabaseNotExists.GenWithStackByArgs(tb.Schema)
+		}
+		t, err := is.TableByName(tb.Schema, tb.Name)
+		if err != nil {
+			return infoschema.ErrTableNotExists.GenWithStackByArgs(tb.Schema, tb.Name)
+		}
```

Comment:
Could we use `getSchemaAndTableByIdent` instead of line3254-3261?

Reply:
Great, thanks.